api/server.go

```diff
@@ -719,22 +719,26 @@ func (s *server) hostsHandler(jc jape.Context) {
 }
 
 func (s *server) searchIDHandler(jc jape.Context) {
-	const maxLen = len(types.Hash256{})
+	const idLen = len(types.Hash256{})
 
 	// get everything after separator if there is one
 	split := strings.Split(jc.PathParam("id"), ":")
 	id, err := hex.DecodeString(split[len(split)-1])
-	if jc.Check("failed to decode hex", err) != nil {
+	if err != nil {
+		jc.Error(errors.New("failed to decode hex"), http.StatusBadRequest)
+		return
+	} else if len(id) < idLen {
+		jc.Error(fmt.Errorf("expected length %d, got %d", idLen, len(id)), http.StatusBadRequest)
 		return
 	}
 
-	trunc := id[:maxLen]
+	trunc := id[:idLen]
 	result, err := s.e.Search(types.Hash256(trunc))
```

Comment:
I'd probably make the search term a string instead since you have to support prefixed keys too.

Reply:
Also please add a test to make sure we actually support it